src/index.ts

```diff
@@ -184,38 +302,50 @@ export class Queue<T> {
 ///   wraps a function so that it executes with context `cx` (or a wrapper around current context if `cx` is falsy).
 ///   The previous context will be restored when the function returns (or throws).  
 ///   returns the wrapped function.
-export function withContext<T>(fn: () => T, cx: any) {
+export function withContext<T>(fn: () => T, cx: any): T {
 	if (!fibers.current) throw new Error('withContext(fn) not allowed outside run()');
-	return _.withContext(fn, cx)();
-}
-
-export function context() {
-	return _.context;
+	const oldContext = globals.context;
+	globals.context = cx || Object.create(oldContext);
+	try {
+		return fn();
+	} finally {
+		globals.context = oldContext;
+	}
 }
 
-// wait variant for streamline.js
-// Callback parameter is typed as any so that f-promise does not re-export streamline-runtime's _ definition.
-export function wait_<T>(arg: (_: any) => T): T {
```

Comment:
I don't know how important this function is.
But, if needed, may be it can move to streamline(-runtime) because they can depend on f-promise now.